### PR TITLE
docs: Nginx with SSL and HTTP/2 is now default

### LINKED ISSUE #3797: Have only NGINX with SSL in mattermost setup
We noticed we still show how to setup Mattermost with NGINX as a reverse proxy without SSL. This can have some serious security implications if people have their instance open to the public internet. 

I want to discuss removing the step without SSL from the docs altogether and only leave the final setup with SSL enabled.

### PR DESCRIPTION
**Pull request by RanadeepPolavarapu**
Resolves #3797.

#### Summary

Updated the docs to remove the older Nginx non-SSL setup. Made Nginx with SSL and HTTP/2 the default. Updated all links to point to this.

Added TLS v1.3 and HTTP/2 Server Push support into the Nginx config.

#### Ticket Link

Fixes #3797.

### CODE REVIEW

**Comment by justinegeffen** [comment]
Hi @RanadeepPolavarapu! Thank you for this, it's incredibly helpful! 

Before I review the docs, I'd like to propose that the new content be added to the existing page as opposed to that page being deleted and a new one created. The reason is that if the current page's link is used anywhere else, readers visiting the page once it's deleted will land on a dead-end. Usually we'd add a blurb such as "this page has been moved to..." however, in this case, the page is nested within the larger guide so it may be a jarring experience. 

I am very open to a deeper discussion and on my side will also look into a redirect - although I don't know for certain that it'll work within the structure of the content. If you have any suggestions they are very welcome! :)

**Comment by RanadeepPolavarapu** [comment]
Hi @justinegeffen,

That's a great point. I have added the original nginx instructions back and added the HTTP/2 SSL config I made on to that document. Now, both documents point to a fully HTTP/2 enabled SSL config.

I have taken the liberty to add TLS v1.3 and HTTP/2 server push support example in the config as well in the documentation to make the config ready for the HTTP/3 QUIC changes about to launch soon in most modern browsers.

Please feel free to let me know if you wish to have something changed.

Best Regards.

**Comment by justinegeffen** [comment]
Thanks so much, @RanadeepPolavarapu, that was really quick! I'll have a chance to review it tomorrow and will touch base with you then. :)

File: source/install/config-proxy-nginx.rst
Comment (justinegeffen) [inline source/install/config-proxy-nginx.rst:24]:
```suggestion
  You will need valid SSL certificates in order for NGINX to pin the certificates properly. Additionally, your browser must have permissions to accept the certificate as a valid CA signed certificate. If you need an example on full configuration with pinning Let's Encrypt, please see the `Nginx HTTP/2 & SSL full configuration guide <https://docs.mattermost.com/install/config-ssl-http2-nginx.html>`__.
```

File: source/install/config-ssl-http2-nginx.rst
Comment (justinegeffen) [inline source/install/config-ssl-http2-nginx.rst:71]:
```suggestion
      server_name mattermost.example.com;
```

File: source/install/config-ssl-http2-nginx.rst
Comment (justinegeffen) [inline source/install/config-ssl-http2-nginx.rst:199]:
```suggestion
This is likely due to a failing cross-origin check. A check is applied for WebSocket code to see if the ``Origin`` header is the same as the host header. If it's not, a 403 error is returned. Open the file ``/etc/nginx/sites-available/mattermost`` as *root* in a text editor and make sure that the host header being set in the proxy is dynamic:
```

**Review by justinegeffen — APPROVED** [review]

**Comment by justinegeffen** [comment]
Minor suggestion but otherwise good to go on my side. Thank you!

**Comment by justinegeffen** [comment]
Adding @DSchalla for visibility.

**Comment by justinegeffen** [comment]
@DSchalla, does this need a security review or can it be merged? :)

**Review by DSchalla — APPROVED** [review]

**Comment by DSchalla** [comment]
Looks good from my end, thank you @RanadeepPolavarapu! I think having Sven's review before merging would make sense to take a look from a support angle.

File: source/install/config-ssl-http2-nginx.rst
Comment (DSchalla) [inline source/install/config-ssl-http2-nginx.rst:148]: @RanadeepPolavarapu Curious why you use a map instead of just use `proxy_set_header Early-Data $ssl_early_data;` directly?

Comment (RanadeepPolavarapu) [reply source/install/config-ssl-http2-nginx.rst:148]: @DSchalla oh I used this for my production server for debugging but yes. This block can be removed for the direct variable.

Comment (justinegeffen) [reply source/install/config-ssl-http2-nginx.rst:148]: Wondering if we should note that in the code and provide the variable too?

Comment (RanadeepPolavarapu) [reply source/install/config-ssl-http2-nginx.rst:148]: Sure will do, I will add a comment noting this.

File: source/install/config-proxy-nginx.rst
Comment (justinegeffen) [inline source/install/config-proxy-nginx.rst:115]:
```suggestion
    # and use the `$ssl_early_data` variable exposed by Nginx directly should you
```

**Review by justinegeffen — APPROVED** [review]

**Review by svelle — APPROVED** [review]
LGTM

**Pull request merged**
